core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitManager.java

```diff
@@ -180,7 +184,13 @@ DistributedTransaction begin(String txId, Isolation isolation, SerializableStrat
         new Snapshot(txId, isolation, strategy, tableMetadataManager, parallelExecutor);
     CrudHandler crud =
         new CrudHandler(storage, snapshot, tableMetadataManager, isIncludeMetadataEnabled);
-    ConsensusCommit consensus = new ConsensusCommit(crud, commit, recovery);
+
+    ConsensusCommit consensus;
+    if (defaultNamespaceName.isPresent()) {
+      consensus = new ConsensusCommit(crud, commit, recovery, defaultNamespaceName.get());
+    } else {
+      consensus = new ConsensusCommit(crud, commit, recovery);
+    }
     getNamespace().ifPresent(consensus::withNamespace);
     getTable().ifPresent(consensus::withTable);
```

Comment:
Any namespace set via the transaction manager (deprecated feature that will be removed in 5.0) by using method such as `DistributedTransactionManager.withNamespace(<namespace>)` and `DistributedTransactionManager.with(<namespace>, <table>)` will have precedence over the default namespace.
Similarly, this also applies to the Storage abstraction and 2pc Transaction manager.